### ISSUE TITLE
failReporter exits with 0 on error

### ISSUE THREAD

**Issue by Hufschmidt** [juanfran/gulp-scss-lint#58]
I'm trying to add gulp-scss-lint as before_prepare hook into cordova (build), but it does not work (quite) yet, since gulp-scss-lint returns with 0 on a lint error instead of 1 (or any other code).

# Example:

## Node:

    var scss  = require('gulp-scss-lint');
    var gulp  = require('gulp');
    
    var paths = [
      './scss/**/*.scss'
    ];
    
    gulp.task('lint', function() {
      return gulp.src(paths)
        .pipe(scss())
        .pipe(scss.failReporter());
    });
    
    gulp.start('lint');

## Scss-File:

    // The path for our ionicons font files, relative to the built CSS in www/css
    $ionicons-font-path: '../lib/ionic/fonts' !default;
    
    throwErrorHere
    
    // Include all of Ionic
    @import 'www/lib/ionic/scss/ionic';

## Output:

    $ node hooks/before_prepare/03_gulp-scss-lint.js && echo "Code: " $?
    [13:01:22] 1 issues found in /home/hufschmt/Workspace/WebView/mILIAS/scss/ionic.app.scss
    [13:01:22] ionic.app.scss:22 [E] Syntax Error: Invalid CSS after "throwErrorHere": expected "{", was ""
    Unhandled rejection Error in plugin 'gulp-scss-lint'
    Message:
        ScssLint failed for: ionic.app.scss
    Code:  0

Is this a bug, or am I doing something wrong? (*Gulp-Scss-Lint version is 0.3.7*)

**Comment by juanfran** [comment]
Thanks for your feedback!

How do you run the gulpfile to get the exit code?

I've tried this and I get `1`

```js
var spawn = require('child_process').spawn,
    lint    = spawn('gulp', ['lint']);

lint.on('close', function (code) {
  console.log('child process exited with code ' + code);
});
```

**Comment by Hufschmidt** [comment]
Hi juanfran,

**./hooks/before_prepare/010_gulp-scss-lint.js:**

    #!/usr/bin/env node

    var lint  = require('gulp-scss-lint');
    var gulp  = require('gulp');

    gulp.task('lint', function() {
      return gulp.src('./src/scss/**/*.scss')
        .pipe(lint())
        .pipe(lint.failReporter());
    });

    gulp.start('lint');

I run this via:

    > nodejs hooks/before_prepare/010_gulp-scss-lint.js

It reports the error in my scss-file, then I check the exit-code via 

    > echo $? 

but I get back a **0**. This procedure does work with gulp-css-lint, eg. this is the script for css linting:

    #!/usr/bin/env node

    var lint = require('gulp-jshint');
    var gulp = require('gulp');

    var paths = [
      './src/js/**/*.js'
    ];

    gulp.task('lint', function() {
      return gulp.src(paths)
        .pipe(lint())
        .pipe(lint.reporter('jshint-stylish'))
        .pipe(lint.reporter('fail'));
    });

    gulp.start('lint');

**Comment by Hufschmidt** [comment]
I did some more investigation on running gulp directly, with

    > gulpfile --gulpfile hooks/before_prepare/010_gulp-scss-lint.js lint

I do get the correct exit-code (*1* on error/warning) on the shell, once I make gulp.src(...) relative to the gulpfile.

This does seem strange though, since I can invoke all other linters (gulp-csslint, gulp-htmlhint, gulp-jshint) using gulp as node module.

**Comment by juanfran** [comment]
I think that I've found the problem but It's very strange and I don't know exactly why... sorry but maybe It will take me some days

**Comment by Hufschmidt** [comment]
Thanks for taking the time. :bowtie:

**Comment by juanfran** [comment]
sorry for the delay, could you try with the 0.3.8 version? thanks

**Comment by Hufschmidt** [comment]
Updated to version 0.3.8, but still the same, I'll add the full output maybe this helps:

	$ node hooks/before_prepare/010_gulp-scss-lint.js 
	[08:34:55] 1 issues found in /home/hufschmt/Workspace/WebView/mILIAS/src/scss/ionic.app.scss
	[08:34:55] ionic.app.scss:43 [E] Syntax Error: Invalid CSS after "...nic/scss/ionic'": expected "{", was ";"
	Unhandled rejection Error in plugin 'gulp-scss-lint'
	Message:
		ScssLint failed for: ionic.app.scss
		
	$ echo $?
	0

and

	$ gulp --gulpfile hooks/before_prepare/010_gulp-scss-lint.js lint
	[08:39:46] Working directory changed to ~/Workspace/WebView/mILIAS/hooks/before_prepare
	[08:39:46] Using gulpfile ~/Workspace/WebView/mILIAS/hooks/before_prepare/010_gulp-scss-lint.js
	[08:39:46] Starting 'lint'...
	[08:39:47] 1 issues found in /home/hufschmt/Workspace/WebView/mILIAS/src/scss/ionic.app.scss
	[08:39:47] ionic.app.scss:43 [E] Syntax Error: Invalid CSS after "...nic/scss/ionic'": expected "{", was ";"
	[08:39:47] 'lint' errored after 266 ms
	[08:39:47] Error in plugin 'gulp-scss-lint'
	Message:
		ScssLint failed for: ionic.app.scss

	$ echo $?
	1

**Comment by juanfran** [comment]
I've test this issue with this https://github.com/juanfran/gulp-scss-lint-test-error before the 0.3.8 I had the wrong exit code, but now works well for me, could you try with this repo and tell me what is different?

thank you!

**Comment by Hufschmidt** [comment]
**TL;DR:** I tested your example and it works, my issue is resolved. :smile:

**Details:**
Extracting your example to a random folder worked both via calling gulp and as node module.

I tried figuring out why this didn't seem to work with my code and copied piece by piece of your example into my working directory, which caused even your example to fail (again).

After this it was clear that something in my working dir was wrong, so I removed all node_modules and reinstalled them with ``npm install``, which low and behold fixed your example (copied into my project folder) and my project...

So my guess is, although I did update via ``npm install gulp-scss-lint`` and even checked version under ``node_modules/gulp-scss-lint/packages.json``, something must have gone wrong, *the problem is fixed in 0.3.8.*

I'll leave the pleasure of closing this issue to you. :smile:

**Comment by juanfran** [comment]
:smile: thanks!
